cmd/controller/api/handlers.go

```diff
@@ -224,7 +224,7 @@ func GetJobHierarchy(w http.ResponseWriter, r *http.Request) {
 
 	var jsonData query.JSONDataWrapper
 	if name, isName := queryParams[query.Name]; isName {
-		jsonData = query.RetrieveJobHierarchy(name[0])
+		jsonData = query.RetrieveResourceHierarchy(query.JobCheck, query.JobType, name[0], query.IsPodFilter)
```

Comment:
Can we abstract all these to a common function/method as well for instance: 
GetResourceHierarchy(query Query) {}
I also suggest trying to create a method maybe. 

struct type Query {
    resourceType String,
    query  Query,
}

func (q Query) GetResourceHierarchy() {}

Reply:
Done